aodntools/timeseries_products/Documentation/Hourly_timeseries.md

```diff
@@ -0,0 +1,157 @@
+
+
+
+### Input file validation
+
+Before proceeding to the aggregation, each input file will be checked to ensure it meets the requirements (as specified above under Inputs). This will prevent errors during the aggregation process. Any input files that fail to meet the requirements will be excluded from the aggregation, and their URL listed in a global attribute rejected_files. 
+
+### Dimensions
+
+The dimensions of the resulting file  are determined as follows:
+
+- `OBSERVATION`:  the total number of observation records, excluding out-of-the-water data, in all input files;
+- `INSTRUMENT`: the number of instruments (i.e. number of files); 
+- `strlen`: a fixed dimension of length 256 for string variables.
+
+
+### Variables
+
+The product will aggregate the following  non-velocity variables (standard/long name in brackets):
+
+`TEMP` (sea_water_temperature);  
+`PSAL` (sea_water_salinity);  
+`CPHL`, `CHLF`, `CHLU` (mass_concentration_of_inferred_chlorophyll_from_relative_fluorescence_units_in_sea_water) - considered as separate variables for now;  
+ `TURB` (sea_water_turbidity);  
+ `DOX1` (mole_concentration_of_dissolved_molecular_oxygen_in_sea_water);   
+`DOX2` (moles_of_oxygen_per_unit_mass_in_sea_water);  
+ `DOXS` (fractional_saturation_of_oxygen_in_sea_water);  
+ `PAR` (downwelling_photosynthetic_photon_flux_in_sea_water).  
+  
+
+
+
+Each of the VoIs are produced by selecting only the “good” and “probably good” data (QC flags 1 and 2), binning the values to a one-hour interval, and concatenating the resulting values into a single file. The resulting variables have dimension OBSERVATION. The VoI’s ancillary_variables, in particular the corresponding quality-control flags, are not included. 
+
+An additional version of the product will include “good”, “probably good” and “non QC performed” data. In this case and for each variable, the percentage of QCed values will be indicated as a variable attribute.
+
+The binning intervals will be one hour long, centred on the hour (i.e. HH:00:00). The binning method will depend on the type of variable being:
+
+| Variables | Method | 
+|-----------|--------|
+| TEMP, CNDC, PRES, PRES_REL,  DEPTH, PSAL, DOXY, DOX, DOX1, DOX1_1, DOX1_2, DOX1_3, DOX2, DOX2_1, DOXS | Mean | 
+| FLU2, CPHL, CHLU, CHLF, TURB, TURBF, PAR | Median | 
+
+The binned `TIME` values are concatenated into a variable `TIME(OBSERVATION)`.  As input files can have overlapping or non-contiguous time coverage, the TIME variable may have duplicate values or gaps.
+
+The binned `DEPTH` values from input files are concatenated into a variable `DEPTH(OBSERVATION)`. If not present, fill values are stored. The variables `PRES` (sea_water_pressure) and `PRES_REL` (sea_water_pressure_due_to_sea_water) will be aggregated in exactly the same way as `DEPTH`. Every aggregated file will contain `DEPTH`, `PRES`, and `PRES_REL`.
```

Comment:
I've suggested this addition in the IP:
> Other variables of interest are treated in a similar way, but will be excluded from the aggregated file if no valid values were found in any of the input files.

Reply:
By the way, I'm not sure that the code actually does make sure that DEPTH, PRES and PRES_REL are kept, even if there are no valid values?

Reply:
actually I'm not checking for the presence of DEPTH or PRES/PRES_REL as I'm assuming that DEPTH should be always there (not necessarily true though). I will include a check in the check_file function, rejecting the file if none of them are present. A solution to missing DEPTH will be replace it with NOMINAL_DEPTH

Reply:
Rejecting an _input_ file with absolutely no depth or pressure information may be a good idea, but here I'm talking about the _output_ file. The doc here (and in the IP) says 
> Every aggregated file will contain `DEPTH`, `PRES`, and `PRES_REL`.

But you recently added code that excludes variables that have only fill values. So that would potentially remove DEPTH, PRES, or PRES_REL as well, no?